docs/table/modify_table.rst

```diff
@@ -164,14 +164,33 @@ the table and replaces existing ones::
    foo     1.0    10.0
    bar     2.0    10.0
 
-:meth:`~astropy.table.Table.update` also takes care of silently :ref:`merging_metadata`::
+When using ``|=``, the other object does not need to be a |Table|, it can be
+anything that can be used for :ref:`construct_table` with a compatible number
+of rows.
 
-  >>> t1.meta
+It is also possible to use ``|`` to return the new |Table| rather than update
+in-place. ``|`` requires the other object to be a |Table|::
+
+  >>> t1 = Table({'name': ['foo', 'bar'], 'val': [0., 0.]}, meta={'n': 2})
+  >>> t2 = Table({'val': [1., 2.], 'val2': [10., 10.]}, meta={'id': 0})
```

Comment:
I'd recommend making `t2` a `QTable` and demonstrating merging both `t1 | t2` and `t2 | t1`. Such examples would document that the output has the same type as the leftmost table to be merged.
```suggestion
  >>> t2 = QTable({'val': [1., 2.], 'val2': [10., 10.]}, meta={'id': 0})
```